### PR TITLE
feat: add security field at the operation level

### PR DESCRIPTION
**Pull request by LokeshRishi**
File                                | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
------------------------------------|---------|----------|---------|---------|-------------------
All files                           |   95.87 |    88.33 |   97.32 |   98.21 |                   
 lib                                |   96.29 |    88.79 |   99.24 |   98.85 |                   
  anonymousNaming.js                |     100 |    95.45 |     100 |     100 | 28                
  asyncapiSchemaFormatParser.js     |     100 |      100 |     100 |     100 |                   
  constants.js                      |     100 |      100 |     100 |     100 |                   
  customValidators.js               |   99.02 |    96.97 |     100 |     100 | 45,513,541-545    
  index.js                          |     100 |      100 |     100 |     100 |                   
  iterators.js                      |   94.64 |     88.1 |   95.65 |   99.04 | 243               
  json-parse.js                     |   95.45 |    66.67 |     100 |   95.45 | 41                
  parser.js                         |   97.35 |    91.07 |     100 |   99.02 | 112               
  utils.js                          |    91.1 |    79.38 |     100 |   96.75 | 83-85,94-96       
 lib/errors                         |      95 |    91.67 |   66.67 |   93.33 |                   
  parser-error.js                   |      95 |    91.67 |   66.67 |   93.33 | 60                
 lib/mixins                         |     100 |      100 |     100 |     100 |                   
  bindings.js                       |     100 |      100 |     100 |     100 |                   
  description.js                    |     100 |      100 |     100 |     100 |                   
  external-docs.js                  |     100 |      100 |     100 |     100 |                   
  specification-extensions.js       |     100 |      100 |     100 |     100 |                   
  tags.js                           |     100 |      100 |     100 |     100 |                   
 lib/models                         |   95.05 |    85.78 |   96.21 |   97.41 |                   
  asyncapi.js                       |   95.83 |     87.1 |   96.77 |   96.58 | 54,274,319,354    
  base.js                           |      80 |       75 |     100 |     100 | 11,21             
  channel-parameter.js              |      90 |       50 |     100 |     100 | 30                
  channel.js                        |   93.55 |    83.33 |     100 |     100 | 75-83             
  components.js                     |     100 |      100 |     100 |     100 |                   
  contact.js                        |     100 |      100 |     100 |     100 |                   
  correlation-id.js                 |     100 |      100 |     100 |     100 |                   
  external-docs.js                  |     100 |      100 |     100 |     100 |                   
  info.js                           |     100 |      100 |     100 |     100 |                   
  license.js                        |     100 |      100 |     100 |     100 |                   
  message-trait.js                  |     100 |      100 |     100 |     100 |                   
  message-traitable.js              |      92 |    66.67 |     100 |     100 | 39-47             
  message.js                        |   93.75 |       80 |     100 |     100 | 33,48-55          
  oauth-flow.js                     |     100 |      100 |     100 |     100 |                   
  operation-security-requirement.js |     100 |      100 |     100 |     100 |                   
  operation-trait.js                |     100 |      100 |     100 |     100 |                   
  operation-traitable.js            |     100 |      100 |     100 |     100 |                   
  operation.js                      |      80 |    66.67 |   88.89 |   86.36 | 18-20             
  publish-operation.js              |      40 |      100 |       0 |      40 | 15-29             
  schema.js                         |     100 |    96.77 |     100 |     100 | 421,437           
  security-scheme.js                |     100 |      100 |     100 |     100 |                   
  server-security-requirement.js    |     100 |      100 |     100 |     100 |                   
  server-variable.js                |   92.31 |       50 |     100 |     100 | 30                
  server.js                         |   94.44 |       50 |     100 |     100 | 69                
  subscribe-operation.js            |      40 |      100 |       0 |      40 | 15-29             
  tag.js                            |     100 |      100 |     100 |     100 |                   
  utils.js                          |     100 |      100 |     100 |     100 |                   

**Description**
We are using AsyncAPI specifications for publishing the subscriber specific details of our topics and associated payloads for our HTTP push use cases.

Our subscribable channels(or topics) are protected by a set of OAuth scopes (authorization code or client credentials) - that a potential subscriber needs to be aware of.

Here is a sample contract.

We are able to accommodate almost every detail of interest to an integrator/subscriber (THANK YOU) - with the exception of being able to call out the oauth scopes necessary for the subscription to the topic - i.e. the security aspect at the channel or rather channel/operation level.

The current specification is perhaps more aligned with a broker based topology - but ours is a pure push use case (to registered webhooks) and we need an ability to call out any security aspects at the channel operation level.

**Related issue(s)**
https://github.com/asyncapi/parser-js/issues/434
https://github.com/asyncapi/spec-json-schemas/pull/189

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -265,6 +265,7 @@
             * [.hasTraits()](#module_@asyncapi/parser+Operation+hasTraits) ⇒ <code>boolean</code>
             * [.messages()](#module_@asyncapi/parser+Operation+messages) ⇒ <code>Array.&lt;Message&gt;</code>
             * [.message()](#module_@asyncapi/parser+Operation+message) ⇒ <code>Message</code>
+            * [.OperationSecurityRequirement](#module_@asyncapi/parser+OperationSecurityRequirement) ⇐ <code>Base</code>
```
Comment (fmvilas) [inline API.md:268]: 🤔 where is this defined?

**Comment by smoya** [comment]
Thanks for the PR @LokeshRishi! would you mind fixing the title to be compliant with [Conventional Commits](https://www.conventionalcommits.org/)? You have an error on it. See https://github.com/asyncapi/parser-js/runs/5698773473?check_suite_focus=true

File: API.md (hunk 1 above)
Comment (smoya) [reply API.md:268]: This doc should be generated automatically iirc. 
Indeed `OperationSecurityRequirement`is not the method you declared, but `Security`.

Comment (LokeshRishi) [reply API.md:268]: Thanks @fmvilas and @smoya 
If its generated automatically, would you suggest reverting my changes?

Comment (smoya) [reply API.md:268]: Would you mind trying `npm run docs`? This command should build the right API doc.

Comment (magicmatatjahu) [reply API.md:268]: The whole PR is ok, just delete the above and I can accept :)

Comment (LokeshRishi) [reply API.md]: @smoya @magicmatatjahu I deleted my changes to the API doc and regenerated it using the above command.

File: test/models/operation_test.js [hunk 2]
```diff
@@ -122,4 +122,15 @@ describe('Operation', function() {
       assertMixinSpecificationExtensionsInheritance(Operation);
     });
   });
+
+  describe('#security()', function() {
+    it('should return an array of security requirements objects', function() {
+      const d = new Operation(js);
+      expect(Array.isArray(d.security())).to.equal(true);
```
Comment (smoya) [inline test/models/operation_test.js:129]: We need to test that the array is not empty. Otherwise, in the case it is empty,  the logic inside the following `forEach` won't ever be executed.

Comment (LokeshRishi) [reply test/models/operation_test.js:129]: @smoya Thanks for the review! I have incorporated the suggestion.

File: test/models/operation_test.js [hunk 3]
```diff
@@ -122,4 +122,16 @@ describe('Operation', function() {
...
+      expect(d.security().length > 0).to.equal(true);
```
Comment (smoya) [inline test/models/operation_test.js:130]: Nitpick: I think you can simplify this with the proper Jest function:
```suggestion
      expect(d.security()).toHaveLength(1);
```

Comment (LokeshRishi) [reply test/models/operation_test.js]: `.toHaveLength(1);` was throwing an error, I have update it to `.to.have.lengthOf(1)`

**Review by smoya — APPROVED** [review]
LGTM!
